src/csprojWriter.ts

```diff
@@ -0,0 +1,121 @@
+import * as path from 'path';
+import * as util from 'util';
+import { promises as fs } from 'fs';
+
+const findUpGlob = require('find-up-glob');
+const xml2js = require("xml2js");
+
+export enum BuildActions {
+    Folder = 'Folder',
+    Compile = 'Compile',
+    Content = 'Content',
+    EmbeddedResource = 'EmbeddedResource',
+    PRIResource = 'PRIResource',
+    Page = 'Page',
+    None = 'None',
+}
+
+export class CsProjWriter {
+    public async getProjFilePath(filePath: string): Promise<string | undefined> {
+        const
+            projItems: string[] = await findUpGlob('*.projitems', { cwd: path.dirname(filePath) }),
+            csProj: string[] = await findUpGlob('*.csproj', { cwd: path.dirname(filePath) });
+
+        if (projItems !== null && projItems.length >= 1) return projItems[0];
+        else if (csProj !== null && csProj.length >= 1) return csProj[0];
+
+        return undefined;
+    }
+
+    public async add(projPath: string, itemPath: string, itemType: BuildActions) {
+        var buildAction = await this.get(projPath, itemPath);
+        if (buildAction !== undefined) await this.remove(projPath, itemPath);
+
+        itemPath = itemPath.replace(path.dirname(projPath) + path.sep, path.extname(projPath) == '.projitems' ? "$(MSBuildThisFileDirectory)" : "");
+
+        const
+            xml = await fs.readFile(projPath, 'utf8'),
+            xmlParser = util.promisify(new xml2js.Parser().parseString),
+            xmlBuilder = new xml2js.Builder();
+
+        let parsedXml = await xmlParser(xml);
+        if (parsedXml === undefined || parsedXml.Project === undefined) return;
```

Comment:
These lines i think repeted in the `add` and `remove` methods. Can't be moved to a method?